File: Dynamic Programming/acmicpc.net/9465/Main.js
const inputFilePath = "/sample.txt";
var input = require("fs")
  .readFileSync(__dirname + inputFilePath)
  // .readFileSync('/dev/stdin')
  .toString()
  .trim()
  .split("\n");

const T = input.shift();

for (let test_case = 0; test_case < T; test_case++) {
  let N = input.shift();
  let a = [[]];
  let line1 = (input.shift().split(" "));
  let line2 = (input.shift().split(" "));
  for (let i = 1; i <= N; i++) {
    a.push([]);
    a[i][1] = parseInt(line1[i - 1]);
    a[i][2] = parseInt(line2[i - 1]);
  }
  let d = [[]];
  d.push([0, a[1][1], a[1][2]])
  for (let j = 2; j <= N; j++) {
    d.push([0, 0, 0]);
    d[j][0] = Math.max(d[j - 1][1], d[j - 1][2]);
    d[j][1] = Math.max(d[j - 1][0], d[j - 1][2]) + a[j][1];
    d[j][2] = Math.max(d[j - 1][0], d[j - 1][1]) + a[j][2];
  }
  console.log(Math.max(Math.max(d[N][0], d[N][1]), d[N][2]));
}

File: Dynamic Programming/acmicpc.net/9465/Main.java
import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.FileInputStream;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.util.StringTokenizer;

class Main {
    static final String inputFilePath = "./Dynamic Programming/acmicpc.net/9465/sample.txt";
    // static final int mod = 1000000000;
    // static final String inputFilePath = "./# format/sample.txt";

    public static void main(String[] args) throws Exception {
        System.setIn(new FileInputStream(inputFilePath));
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        int T = Integer.parseInt(br.readLine());
        for (int i = 0; i < T; i++) {
            int N = Integer.parseInt(br.readLine());
            int a[][] = new int[N + 1][3];
            int d[][] = new int[N + 1][3];
            StringTokenizer line1 = new StringTokenizer(br.readLine());
            StringTokenizer line2 = new StringTokenizer(br.readLine());
            for (int j = 1; j <= N; j++) {
                a[j][1] = Integer.parseInt(line1.nextToken());
                a[j][2] = Integer.parseInt(line2.nextToken());
            }

            d[1][0] = 0;
            d[1][1] = a[1][1];
            d[1][2] = a[1][2];
            for (int j = 2; j <= N; j++) {
                d[j][0] = Math.max(d[j - 1][1], d[j - 1][2]) + a[j][0];
                d[j][1] = Math.max(d[j - 1][0], d[j - 1][2]) + a[j][1];
                d[j][2] = Math.max(d[j - 1][0], d[j - 1][1]) + a[j][2];
            }
            System.out.println(Math.max(Math.max(d[N][0], d[N][1]), d[N][2]));
        }
    }
}